### PR TITLE
Decouple filament minimum print speed from overhang slowdown

### PR DESCRIPTION
**Pull request by igiannakas**
Amending default behaviour to make it consistent with Prusa slicer.

Currently the minimum filament print speed caps the reduced print speed resulting from overhang slowdown. 

This is undesirable in some cases, and specifically where you want a high minimum print speed for the filament to reduce wall artefacts resulting from slowing down for layer time too much, but want to keep the reduced speeds in the overhanging sections.

It is also a "silent" override to the user set print speeds in the overhang sections. 

(ignore the different speeds between Prusa & Orca as I didn't match the profiles - the principle is what I'm trying to show below)
**Prusa:**
Slows down overhangs to the min overhang speed irrespective of the filament min print speed which is set to 100mm/sec for illustration purposes
image

**Before:**
Filament print speed of 100mm/sec caps the slowdowns to that (illustration purposes - you shouldn't run 100mm/sec filament min print speeds for cooling!)
image

**After:**
Slowdown for overhangs are respected and the min print speed also respected in areas where there is no overhang.
image

The reason for this PR is I'm trying to maximise external wall surface finish on my HQ profiles and want to minimise variance to the print speeds for the external walls. In my HQ profile I set external wall speed to 40mm/sec and want to set minimum print speed to 40mm/sec to make sure the external walls stay at the same speed when slowing down for layer time. However what this did is worsen my overhangs as I cannot slow them down to 20mm/sec where they print best especially for curled edges. Hence the need to decouple the two. 

If we want a global filament minimum print speed (not related to cooling) we could add a different option and make it so that no feature prints slower than this (currently it is limited to the overhangs only).

### CODE REVIEW

**Comment by Marius955** [comment]
One question, your comments could solve this problem that I have in some parts, which I assume is due to the speed changes.
photo_2024-01-31_21-43-11

**Comment by igiannakas** [comment]
> One question, your comments could solve this problem that I have in some parts, which I assume is due to the speed changes. photo_2024-01-31_21-43-11

It depends what else is happening there. It looks like your PA is probably slightly wrong for the tan filament, or the retraction settings are a bit off.

**Comment by discip** [comment]
@igiannakas
The tan color seems to actually be painted on, instead of being another filament. 😊

**Comment by igiannakas** [comment]
> @igiannakas The tan color seems to actually be painted on, instead of being another filament. 😊

🤣 I need more coffee! Indeed you’re right!

this PR won’t address this. This is extrusion issue not a speed issue that this PR is aiming to tweak.

File: src/libslic3r/PrintConfig.cpp [hunk 1]
```diff
@@ -2941,7 +2941,8 @@ def = this->add("filament_loading_speed", coFloats);
 
     def = this->add("slow_down_min_speed", coFloats);
     def->label = L("Min print speed");
-    def->tooltip = L("The minimum printing speed for the filament when slow down for better layer cooling is enabled, when printing overhangs and when feature speeds are not specified explicitly.");
+    def->tooltip = L("The minimum printing speed that the printer will slow down to to attempt to maintain the minimum layer time "
+                     "above, when slow down for better layer cooling is enabled.");
```
Comment (SoftFever) [inline src/libslic3r/PrintConfig.cpp:2945]: Good change.
It expain the parameter more precisely now.

**Comment by SoftFever** [comment]
Thank you for your patience. :)

After giving it some thought and reflecting on my own printing experience, I questioned the utility of using it as a global minimum speed threshold too. 
My conclusion is that it hasn't been particularly useful. 
Therefore, I agree that we should limit its application to serving as a threshold for minimum layer time slowdown only.

**Review by SoftFever — APPROVED** [review]
LGTM

**Pull request merged**